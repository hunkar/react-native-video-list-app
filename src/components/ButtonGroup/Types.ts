import {IconNode} from 'react-native-elements/dist/icons/Icon';
import {StyleProp, ViewStyle} from 'react-native';

//TODO Other styles can be customized. Can be added in the future.
/**
 * title: text of button
 * icon: icon of button
 * onPress: press event of button
 */
export type ButtonProps = {
  title: string;
  icon?: IconNode;
  onPress?: (() => void) | undefined;
};

/**
 * containerStyle: Styles for group container
 * buttons: array of ButtonProps
 */
export type ButtonGroupProps = {
  containerStyle?: StyleProp<ViewStyle>;
  buttons: ButtonProps[];
};
