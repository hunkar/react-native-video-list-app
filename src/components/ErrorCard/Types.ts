/**
 * title: Text of error.
 * onRetry: function that will be called on retry button clicked.
 */
export type ErrorCardProps = {
  title: string;
  onRetry?: () => void;
};
