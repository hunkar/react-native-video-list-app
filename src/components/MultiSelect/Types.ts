/**
 * id: id of item,
 * name: item name that will be shown
 * isSelected: you can set select status of item
 * value: extra value for item. For external use.
 */
export type MultiSelectItemProps = {
  id: string;
  name: string;
  isSelected?: boolean;
  value?: string | number | boolean;
};

/**
 * items: array of MultiSelectItemProps
 * selectedItems: id array of selected items
 * onChange: trigger function with selected items
 */
export type MultiSelectProps = {
  items: MultiSelectItemProps[];
  selectedItems?: string[];
  onChange?: (item: MultiSelectItemProps[]) => void;
};
