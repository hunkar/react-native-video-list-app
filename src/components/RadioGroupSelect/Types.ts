/**
 * id: item id
 * name: item name that will be show on the list
 * isSelected: item can be selectable with this prop
 * value: any value for external use.
 */
export type RadioGroupSelectItemProps = {
  id: string;
  name: string;
  isSelected?: boolean;
  value?: string | number | boolean;
};

/**
 * items: array of RadioGroupSelectItemProps items
 * selectedItem: id of selected item
 * onChange: trigger with selected item on change
 */
export type RadioGroupSelectProps = {
  items: RadioGroupSelectItemProps[];
  selectedItem?: string;
  onChange?: (item: RadioGroupSelectItemProps) => void;
};
