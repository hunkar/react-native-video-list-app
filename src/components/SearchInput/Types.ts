/**
 * placeholder: placeholder for searchBar
 * onChangeText: event for text change
 * changeTimeout: letter input frequency waiting range in ms 
 */
export type SearchInputProps = {
  placeholder?: string;
  onChangeText?: (text: string) => void;
  changeTimeout?: number;
};
