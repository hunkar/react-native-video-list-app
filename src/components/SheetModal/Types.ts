/**
 * title: title of modal
 * children: any ReactNode for modal child
 */
export type SheetModalProps = {
  title: string;
  children: React.ReactNode;
};
