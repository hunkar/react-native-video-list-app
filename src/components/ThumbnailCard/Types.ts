/**
 * id: Id of music.
 * title: Main title for singer name.
 * subTitle: Sub title for music name.
 * imageUrl: Url of video image.
 */
export type ThumbnailCardProps = {
  id: string;
  title: string;
  subTitle: string;
  imageUrl: string;
};
