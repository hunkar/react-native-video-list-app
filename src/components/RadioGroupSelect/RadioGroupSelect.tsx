import React, {useState} from 'react';
import {FlatList, StyleSheet} from 'react-native';
import {Icon, ListItem} from 'react-native-elements';
import {RadioGroupSelectItemProps, RadioGroupSelectProps} from './Types';

/**
 * This component is row item of radio group.
 * Include row item click, style logics.
 *
 * @param param0 properties of item
 * @returns ReactNode
 */
const Item = ({
  id,
  name,
  isSelected,
  onPress,
}: {
  id: string;
  name: string;
  isSelected: boolean;
  onPress?: (() => void) | undefined;
}) => (
  <ListItem
    testID={`filter_item_${id}`}
    key={id}
    containerStyle={{
      backgroundColor: isSelected ? '#d6d6d6' : 'white',
    }}
    onPress={onPress}
    hasTVPreferredFocus={undefined}
    tvParallaxProperties={undefined}>
    <ListItem.Content style={styles.itemTextContent}>
      <Icon
        testID={`filter_item_icon_${id}`}
        name={isSelected ? 'radio-button-on' : 'radio-button-off'}
        style={styles.itemIcon}
        tvParallaxProperties={undefined}
      />
      <ListItem.Title testID={`filter_item_text_${id}`}>{name}</ListItem.Title>
    </ListItem.Content>
  </ListItem>
);

/**
 * Custom radio group select. You can toggle from row items.
 *
 * @param props RadioGroupSelectProps
 * @returns ReactNode
 */
const RadioGroupSelect = (props: RadioGroupSelectProps) => {
  const {items, onChange} = props;

  const [selected, setSelected] = useState<string>(props.selectedItem || '');

    /**
   * Render row item. Include row click logic for  radio group select component.
   *
   * @param param0 MultiSelectItemProps
   * @returns
   */
  const renderItem = ({item}: {item: RadioGroupSelectItemProps}) => (
    <Item
      onPress={() => {
        if (onChange) {
          onChange(item);
        }
        setSelected(item.id);
      }}
      isSelected={
        item.isSelected || (selected && item.id === selected) || false
      }
      {...item}
    />
  );

  return (
    <FlatList
      style={styles.flatList}
      data={items}
      renderItem={renderItem}
      keyExtractor={(item: RadioGroupSelectItemProps) => item.id}
    />
  );
};

const styles = StyleSheet.create({
  itemTextContent: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'flex-start',
    justifyContent: 'flex-start',
  },
  itemIcon: {
    marginRight: 15,
  },
  flatList: {
    height: '100%',
  },
});

export default RadioGroupSelect;
