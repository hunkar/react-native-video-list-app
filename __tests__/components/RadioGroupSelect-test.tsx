import renderer, {act} from 'react-test-renderer';
import React from 'react';
import { RadioGroupSelectItemProps, RadioGroupSelectProps } from '../../src/components/RadioGroupSelect/Types';
import RadioGroupSelect from '../../src/components/RadioGroupSelect/RadioGroupSelect';

test('snapshot equility', async () => {
  const firstElement: RadioGroupSelectItemProps = {
    id: '1',
    name: 'First Element',
    value: 1,
  };

  const secondElement: RadioGroupSelectItemProps = {
    id: '2',
    name: 'Second Element',
    value: 1,
  };

  const mockProps: RadioGroupSelectProps = {
    items: [firstElement, secondElement],
    onChange: jest.fn(),
  };

  const component = renderer.create(<RadioGroupSelect {...mockProps} />);

  expect(component.toJSON()).toMatchSnapshot();
});

test('renders elements', async () => {
  const firstElement: RadioGroupSelectItemProps = {
    id: '1',
    name: 'First Element',
    value: 1,
  };

  const secondElement: RadioGroupSelectItemProps = {
    id: '2',
    name: 'Second Element',
    value: 1,
  };

  const mockProps: RadioGroupSelectProps = {
    items: [firstElement, secondElement],
    onChange: jest.fn(),
  };

  const component = renderer.create(<RadioGroupSelect {...mockProps} />);

  const firstElementTextComponent = component.root.findByProps({
    testID: 'filter_item_text_1',
  }).props;

  const secondElementTextContainer = component.root.findByProps({
    testID: 'filter_item_text_2',
  }).props;

  expect(firstElementTextComponent.children).toEqual(firstElement.name);
  expect(secondElementTextContainer.children).toEqual(secondElement.name);
});

test('Clicked item should be trigger on change as selected.', async () => {
  const firstElement: RadioGroupSelectItemProps = {
    id: '1',
    name: 'First Element',
    value: 1,
  };

  const secondElement: RadioGroupSelectItemProps = {
    id: '2',
    name: 'Second Element',
    value: 1,
  };

  const mockProps: RadioGroupSelectProps = {
    items: [firstElement, secondElement],
    selectedItem: firstElement.id,
    onChange: jest.fn(),
  };

  const component = renderer.create(<RadioGroupSelect {...mockProps} />);

  const firstElementComponent = component.root.findByProps({
    testID: 'filter_item_1',
  }).props;

  act(() => {
    firstElementComponent.onPress();
  });

  expect(mockProps.onChange).toBeCalledTimes(1);
  expect(mockProps.onChange).toBeCalledWith(firstElement);
});

test('Second item should be trigger on change after first', async () => {
  const firstElement: RadioGroupSelectItemProps = {
    id: '1',
    name: 'First Element',
    value: 1,
  };

  const secondElement: RadioGroupSelectItemProps = {
    id: '2',
    name: 'Second Element',
    value: 1,
  };

  const mockProps: RadioGroupSelectProps = {
    items: [firstElement, secondElement],
    selectedItem: firstElement.id,
    onChange: jest.fn(),
  };

  const component = renderer.create(<RadioGroupSelect {...mockProps} />);

  const secondElementComponent = component.root.findByProps({
    testID: 'filter_item_2',
  }).props;

  act(() => {
    secondElementComponent.onPress();
  });

  expect(mockProps.onChange).toBeCalledTimes(1);
  expect(mockProps.onChange).toBeCalledWith(secondElement);
});

test('Selected item should have selected icon.', async () => {
  const firstElement: RadioGroupSelectItemProps = {
    id: '1',
    name: 'First Element',
    value: 1,
  };

  const secondElement: RadioGroupSelectItemProps = {
    id: '2',
    name: 'Second Element',
    value: 1,
  };

  const mockProps: RadioGroupSelectProps = {
    items: [firstElement, secondElement],
    selectedItem: firstElement.id,
    onChange: jest.fn(),
  };

  const component = renderer.create(<RadioGroupSelect {...mockProps} />);

  const iconComponent = component.root.findByProps({
    testID: 'filter_item_icon_1',
  }).props;

  expect(iconComponent.name).toEqual('radio-button-on');
});
