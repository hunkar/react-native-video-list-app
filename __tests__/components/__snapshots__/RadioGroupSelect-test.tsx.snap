// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot equility 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "id": "1",
        "name": "First Element",
        "value": 1,
      },
      Object {
        "id": "2",
        "name": "Second Element",
        "value": 1,
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={
    Object {
      "height": "100%",
    }
  }
  viewabilityConfigCallbackPairs={Array []}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        testID="filter_item_1"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderColor": "#bcbbc1",
              "flexDirection": "row",
              "padding": 14,
            }
          }
          testID="padView"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "flex-start",
                  "flex": 1,
                  "justifyContent": "center",
                },
                Object {
                  "alignItems": "flex-start",
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                },
              ]
            }
            theme={
              Object {
                "colors": Object {
                  "black": "#242424",
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": Object {
                    "android": Object {
                      "error": "#f44336",
                      "grey": "rgba(0, 0, 0, 0.54)",
                      "primary": "#2196f3",
                      "searchBg": "#dcdce1",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "default": Object {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "ios": Object {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "web": Object {
                      "error": "#ff190c",
                      "grey": "#393e42",
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#ca71eb",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ca71eb",
                  "success": "#52c41a",
                  "warning": "#faad14",
                  "white": "#ffffff",
                },
              }
            }
          >
            <View
              style={
                Object {
                  "overflow": "hidden",
                }
              }
            >
              <View
                style={
                  Object {
                    "marginRight": 15,
                  }
                }
                testID="filter_item_icon_1"
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "color": "#242424",
                          "fontSize": 24,
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                        Object {
                          "fontFamily": "Material Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                    testID="iconIcon"
                  >
                    î ¶
                  </Text>
                </View>
              </View>
            </View>
            <Text
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#242424",
                  "fontSize": 17,
                }
              }
              testID="filter_item_text_1"
            >
              First Element
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        testID="filter_item_2"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderColor": "#bcbbc1",
              "flexDirection": "row",
              "padding": 14,
            }
          }
          testID="padView"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "flex-start",
                  "flex": 1,
                  "justifyContent": "center",
                },
                Object {
                  "alignItems": "flex-start",
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                },
              ]
            }
            theme={
              Object {
                "colors": Object {
                  "black": "#242424",
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": Object {
                    "android": Object {
                      "error": "#f44336",
                      "grey": "rgba(0, 0, 0, 0.54)",
                      "primary": "#2196f3",
                      "searchBg": "#dcdce1",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "default": Object {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "ios": Object {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "web": Object {
                      "error": "#ff190c",
                      "grey": "#393e42",
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#ca71eb",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ca71eb",
                  "success": "#52c41a",
                  "warning": "#faad14",
                  "white": "#ffffff",
                },
              }
            }
          >
            <View
              style={
                Object {
                  "overflow": "hidden",
                }
              }
            >
              <View
                style={
                  Object {
                    "marginRight": 15,
                  }
                }
                testID="filter_item_icon_2"
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    selectable={false}
                    style={
                      Array [
                        Object {
                          "color": "#242424",
                          "fontSize": 24,
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                        Object {
                          "fontFamily": "Material Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                    testID="iconIcon"
                  >
                    î ¶
                  </Text>
                </View>
              </View>
            </View>
            <Text
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#242424",
                  "fontSize": 17,
                }
              }
              testID="filter_item_text_2"
            >
              Second Element
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
